src/play-item.js

```diff
@@ -38,15 +38,18 @@ const playItem = (player, delay, item) => {
   player.trigger('beforeplaylistitem', item);
   player.poster(item.poster || '');
   player.src(item.sources);
-  clearTracks(player);
-  (item.textTracks || []).forEach(player.addRemoteTextTrack.bind(player));
-  player.trigger('playlistitem', item);
 
-  if (replay) {
-    player.play();
-  }
+  player.ready(() => {
+    clearTracks(player);
```

Comment:
wouldn't we want to clear tracks before switching?

Reply:
Yes